### PR TITLE
Profiles: Limit opensea api requests

### PR DESCRIPTION
**Pull request by benisgold**
Fixes RNBW-3743

## What changed (plus any additional context for devs)
- limited opensea api requests

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/handlers/opensea-api.ts [hunk 1]
```diff
@@ -83,14 +88,16 @@ export const apiGetAccountUniqueToken = async (
...
+    const dataTask = async () =>
+      await rainbowFetch(isPolygon ? urlV2 : url, {
+        headers: {
+          'Accept': 'application/json',
+          'X-Api-Key': NFT_API_KEY,
+        },
+        method: 'get',
+        timeout: 10000, // 10 secs
+      });
+    const { data } = await queue.add(dataTask);
```
Comment (jxom) [inline src/handlers/opensea-api.ts:100]: can we just inline the async function instead of defining `dataTask`?

File: package.json [hunk 2]
```diff
@@ -154,6 +154,7 @@
     "mnemonist": "0.38.1",
     "multiformats": "9.6.2",
     "nanoid": "3.2.0",
+    "p-queue": "7.2.0",
```
Comment (estebanmino) [inline package.json:157]: dont want to block this PR and this dep seems safe, but it's strictly needed?

**Review by estebanmino — APPROVED** [review]

File: src/handlers/opensea-api.ts (hunk 1 above)
Comment (benisgold) [reply src/handlers/opensea-api.ts]: 🫡

File: package.json (hunk 2 above)
Comment (benisgold) [reply package.json:157]: hmm i could hypothetically use setTimeout but this way is more effective

Comment (jxom) [reply package.json:157]: yeah, think it should be okay, defs dont wanna implement queueing logic from scratch 😝

File: src/handlers/opensea-api.ts [hunk 3]
```diff
@@ -17,6 +18,8 @@ import logger from 'logger';
 export const UNIQUE_TOKENS_LIMIT_PER_PAGE: number = 50;
 export const UNIQUE_TOKENS_LIMIT_TOTAL: number = 2000;
 
+const queue = new PQueue({ interval: 1000, intervalCap: 10 });
```
Comment (jxom) [inline src/handlers/opensea-api.ts:21]: maybe we should add a quick comment here to explain why we are needing this?

**Review by jxom — APPROVED** [review]

**Pull request merged**